storage/entities/transaction.js

```diff
@@ -374,7 +380,17 @@ class Transaction extends BaseEntity {
 				)
 			);
 
-			const groupedTransactions = _.groupBy(transactions, 'type');
+			// Not all transactions of type 0 might have an `asset` so remove the ones without `asset` before storing related data
+			const needSubtransactions = transactions.filter(
+				transaction =>
+					transaction.type !== 0 ||
+					(transaction.type === 0 &&
+						transaction.asset &&
+						transaction.asset.data)
+			);
```

Comment:
If any filtering required baed on transaction type, it should be done under `switch` below.

Reply:
Updated based on suggestion.